tests/queries/0_stateless/02907_backup_restore_flatten_nested.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} -nm --query "
+drop table if exists test;
+set flatten_nested = 0;
+create table test (test Array(Tuple(foo String, bar Float64))) ENGINE = MergeTree() ORDER BY tuple();
```

Comment:
We will soon only flatten the explicitly stated Nested type: https://github.com/ClickHouse/ClickHouse/pull/56132

Reply:
Ok, but the same problem with backup restore can be even with explicitly stated Nested type

Reply:
My suggestion was to change it to something like this for this test to make sense even after the change I mentioned.
```suggestion
create table test (test Nested(foo String, bar Float64)) ENGINE = MergeTree() ORDER BY tuple();
```